tests/Integration/Behaviour/Features/Context/Domain/AttributeFeatureContext.php

```diff
@@ -0,0 +1,219 @@
+
+use Behat\Gherkin\Node\TableNode;
+use PHPUnit\Framework\Assert;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\Attribute\Command\AddAttributeCommand;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\Attribute\Command\DeleteAttributeCommand;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\Attribute\Command\EditAttributeCommand;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\Attribute\Exception\AttributeConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\Attribute\Exception\AttributeNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\Attribute\Query\GetAttributeForEditing;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\Attribute\QueryResult\EditableAttribute;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\Attribute\ValueObject\AttributeId;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\Exception\AttributeGroupConstraintException;
+use Tests\Integration\Behaviour\Features\Context\Util\NoExceptionAlthoughExpectedException;
+
+class AttributeFeatureContext extends AbstractDomainFeatureContext
+{
+    /**
+     * @When I create attribute :reference with specified properties:
+     */
+    public function createAttribute(string $reference, TableNode $node): void
+    {
+        $properties = $this->localizeByRows($node);
+
+        $attributeGroupId = $this->referenceToId($properties['attribute_group']);
+        $attributeId = $this->createAttributeUsingCommand(
+            $attributeGroupId,
+            $properties['value'],
+            $properties['color'],
+            $this->referencesToIds($properties['shopIds'])
+        );
+
+        $this->getSharedStorage()->set($reference, $attributeId->getValue());
+    }
+
+    /**
+     * @When I create attribute :reference with invalid color I should get an exception:
+     */
+    public function createAttributeWithInvalidColor(string $reference, TableNode $node): void
+    {
+        $caughtException = null;
+        $properties = $this->localizeByRows($node);
+        $attributeGroupId = $this->referenceToId($properties['attribute_group']);
+
+        try {
+            $this->createAttributeUsingCommand(
+                $attributeGroupId,
+                $properties['value'],
+                $properties['color'],
+                $this->referencesToIds($properties['shopIds'])
+            );
+        } catch (AttributeConstraintException $e) {
+            $caughtException = $e;
+        }
+
+        Assert::assertNotNull(
+            $caughtException,
+            sprintf('Creating an attribute with invalid color %s should trigger an exception', $properties['color'])
+        );
+        Assert::assertEquals(
+            $caughtException->getCode(),
+            AttributeConstraintException::INVALID_COLOR,
+            sprintf('The thrown exception for an invalid color should have the code %d', AttributeConstraintException::INVALID_COLOR)
+        );
+    }
```

Comment:
You don't need a dedicated endpoint to handle the fault creation/editionn, for exception checking we have a system to temporarily store the caught exception thanks to the `setLastException` method in the regular create/edit endpoint like here:
https://github.com/PrestaShop/PrestaShop/blob/d11bf58f17a89b237f30ff7173f4b5c6b4015ded/tests/Integration/Behaviour/Features/Context/Domain/Product/UpdateProductFeatureContext.php#L99

This allows you to then add a dedicated behat endpoint only to check that this exception is the expected one like here:
https://github.com/PrestaShop/PrestaShop/blob/8b69a5a2cfcd38378bee9abfc6a601993bb34aa3/tests/Integration/Behaviour/Features/Context/Domain/Product/SpecificPriceContext.php#L381-L384